### ISSUE TITLE
Datadog Agent on macOS crashed with: "Runtime: Split stack overflow"

### ISSUE THREAD

**Issue by goodspark** [DataDog/datadog-agent#6538]
```
{
	"StandardOutPath" = "/var/log/datadog/launchd.log";
	"LimitLoadToSessionType" = "Aqua";
	"StandardErrorPath" = "/var/log/datadog/launchd.log";
	"Label" = "com.datadoghq.agent";
	"OnDemand" = true;
	"LastExitStatus" = 9;
	"PID" = 10050;
	"Program" = "/opt/datadog-agent/bin/agent/agent";
	"ProgramArguments" = (
		"/opt/datadog-agent/bin/agent/agent";
		"run";
	);
};
```

**Describe what happened:**
The Datadog agent crashed over night.

**Describe what you expected:**
To not crash.

**Steps to reproduce the issue:**
Not sure yet. I just had the agent running on various macOS machines overnight and almost all of them crashed.

**Additional environment details (Operating System, Cloud provider, etc):**
Mac OS 10.15.7, build 19H2
Mac Mini
Intel(R) Core(TM) i7-8700B CPU @ 3.20GHz
16GB RAM
Datadog Agent 7.23.0 - Commit: 3cc240ddd - Serialization version: v4.44.0 - Go version: go1.14.7

I've attached the contents of /var/log/datadog/launchd.log from one of the mac machines.
[launchd.log](https://github.com/DataDog/datadog-agent/files/5356168/launchd.log)

**Comment by mx-psi** [comment]
Hi @goodspark, thanks for reporting this issue and issue #6539.

We are currently investigating both issues; they seem to affect `v7.23.0`/`v6.23.0` of the Datadog Agent. To investigate this further, we would like you to [send a flare](https://docs.datadoghq.com/agent/troubleshooting/send_a_flare/?tab=agentv6v7) from one of the Agents affected by this and another one for issue #6539.

Until we release a fix we have reverted the latest MacOS installer to 7.22.0 and we recommend you to use that version (please send the flare before changing versions).

**Comment by mx-psi** [comment]
A small update: we have identified this as a Go macOS runtime bug, see golang/go#39079. We are waiting on a bugfix release of Go 1.14 that will help us tackle this problem. I will keep you posted with further updates.

**Comment by mx-psi** [comment]
Hi @goodspark, the recently released Agent version 7.25.0 should fix this issue and issue #6539 since we upgraded Go (see #6942). Feel free to comment or reopen if you find the issue again after upgrading.
